src/drivers/dw/dma.c

```diff
@@ -1411,6 +1437,56 @@ static int dw_dma_remove(struct dma *dma)
 	return 0;
 }
 
+static int dw_dma_in_data_size(struct dma_chan_data *chan)
+{
+	int size = 0;
+	uint32_t read_ptr = chan->ptr_data.current_ptr;
+	uint32_t write_ptr = dw_read(chan->id.dma, DW_DAR(chan->id.channel));
+
+	size = (int)write_ptr - (int)read_ptr;
```

Comment:
isn't this what intptr_t was invented for?

Reply:
@plbossart you mean for read_ptr and write_ptr? For size it would be rather ssize_t? Are type-casts really needed?

Reply:
@lyakh similar with int32_t there is intptr_t. The purpose of this pointer typedef is to be able to substract pointers.
It is similar with the cast in your patch but in a more fancy way.

https://github.com/thesofproject/newlib-xtensa/blob/xtensa/newlib/libc/include/stdint.h#L247

Reply:
Thanks, I know what intptr_t is for. I meant - shouldn't intptr_t be used for read_ptr and write_ptr, not for "size"